### PR TITLE
Move push to a new NFCWriter object

### PR DESCRIPTION
**Pull request by kenchris**
The easy part of issue #152

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1182,7 +1190,7 @@
     <p>
       The <dfn>NFCMessage.url</dfn>
       property represents the <a>Web NFC Id</a> of a received
-      <a>Web NFC message</a>. When used in the <code><a>NFC.push</a>()</code>
+      <a>Web NFC message</a>. When used in the <code><a>NFCWriter.push</a>()</code>
```
Comment (littledan) [inline index.html:1193]: Nit: Consider writing something like NFCWriter.prototype.push (fine to do this in another PR)

Comment (kenchris) [reply index.html:1193]: Sure. I hope that works with respec :-)

File: index.html [hunk 2]
```diff
@@ -1564,7 +1580,7 @@ <h2>The <dfn>NFCRecordType</dfn> enum</h2>
       Set <var>nfc</var>.[[\suspended]] to <code>true</code>.
     </li>
     <li>
-      For each pending <a>NFC.push</a>:
+      For each pending <a>NFCWriter.push</a>:
```
Comment (littledan) [inline index.html:1583]: I guess this is talking about making the NFCWriter object disappear, and we're doing this for each push pending on the NFCWriter object, right? (Since presumably this will be called for all the NFCWriter objects.)

We should probably have a little layering change in HTML's [unloading document cleanup steps](https://html.spec.whatwg.org/#unloading-document-cleanup-steps) to let all platform objects register "make disappear" steps. (And, going crazy with it, maybe we should have a JS API for this as well?)

(Doesn't need to be clarified in this PR)

File: index.html [hunk 3]
```diff
@@ -1448,17 +1456,25 @@ <h2>The <dfn>NFCRecordType</dfn> enum</h2>
   <pre class="idl">
     typedef (DOMString or ArrayBuffer or NFCMessage) NFCMessageSource;
 
+    [Constructor(), SecureContext, Exposed=Window]
+    interface NFCWriter {
+      Promise&lt;void&gt; push(NFCMessageSource message, optional NFCPushOptions options);
+    };
+
     [SecureContext]
```
Comment (littledan) [inline index.html:1464]: The NFC interface should either have `[Exposed=Window]` or `[NoInterfaceObject]` (I guess the latter?). Anyway, the extended attributes for NFCWriter LGTM, and I guess NFC will go away when you add NFCReader.

File: index.html [hunk 4]
```diff
@@ -1448,17 +1456,25 @@ <h2>The <dfn>NFCRecordType</dfn> enum</h2>
...
     interface NFC {
-      Promise&lt;void&gt; push(NFCMessageSource message, optional NFCPushOptions options);
       Promise&lt;void&gt; watch(MessageCallback callback, optional NFCWatchOptions options);
     };
 
     callback MessageCallback = void (NFCMessage message);
   </pre>
+  <p>
+    The <dfn>NFCWriter</dfn> is an object used for writing data to NFC devices
```
Comment (littledan) [inline index.html:1472]: Nit: Instances of NFCWriter are used to write data to NFC devices (this is not a singleton class)

**Review by littledan — APPROVED** [review]
This looks fine to land; all comments I have are separate things to look into changing.

File: index.html (hunk 2 above)
Comment (littledan) [reply index.html:1583]: Added https://github.com/w3c/web-nfc/pull/178/commits/459a51a551b309278171f014f32ed9c2b39f4466 to clarify these things (I had previously missed the connection below, and seems like we don't need a layering change).
